### PR TITLE
[Bug Fix] LPS-142911 Remove support for regex, use String to match

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

**Comment by yuhai** [comment]
ci:test

**Comment by dantewang** [comment]
@zhangyan0701 Could you review the test results? The number of ci-test-default failures seems unusually high.

**Comment by zhangyan0701** [comment]
The ci default test results were not imported to Testray somehow, will retrigger the tests.

**Comment by zhangyan0701** [comment]
ci:test

**Comment by tinatian** [comment]
@vicnate5 Please help me scan test results.

Thanks.
Tina

File: portal-impl/test/unit/com/liferay/portal/util/HttpImplTest.java [hunk 1]
```diff
@@ -76,7 +78,7 @@ public void appendAssertClasses(List<Class<?>> assertClasses) {
 			LiferayUnitTestRule.INSTANCE);
 
 	@BeforeClass
-	public static void setUpClass() {
+	public static void setUpClass() throws Exception {
```
Comment (tinatian) [inline portal-impl/test/unit/com/liferay/portal/util/HttpImplTest.java:81]: @yuhai Why this change is needed?

Tina

File: portal-impl/src/com/liferay/portal/util/HttpImpl.java [hunk 2]
```diff
@@ -711,11 +702,16 @@ public boolean isNonProxyHost(String host) {
...
 		}
 
-		if (_nonProxyHostsPattern != null) {
-			Matcher matcher = _nonProxyHostsPattern.matcher(host);
+		if (_nonProxyHosts != null) {
+			for (String nonProxyHost : _nonProxyHosts) {
+				if (nonProxyHost.equals(host) ||
+					(nonProxyHost.contains(StringPool.STAR) &&
+					 StringUtil.wildcardMatches(
+						 host, nonProxyHost, CharPool.UNDERLINE, CharPool.STAR,
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portal/util/HttpImpl.java:710]: @yuhai The test did not cover the "_"  and "/" logic here.

**Comment by tinatian** [comment]
@yuhai Please see my comments and resend.

Thanks.
Tina.

**Comment by tinatian** [comment]
ci:close
